From e9ea6a27e14e1a5de12c9755bf63ad6366fcbe22 Mon Sep 17 00:00:00 2001
From: Adrian Dombeck <adrian.dombeck@code-intelligence.com>
Date: Fri, 16 Dec 2022 12:14:28 +0100
Subject: [PATCH] Link libcap statically

... to avoid it as a runtime dependency.

This commit contains the changes done by:

    sed -i "s/-lcap/-l:libcap.a/g" Makefile
---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 4759736..f505e84 100644
--- a/Makefile
+++ b/Makefile
@@ -103,7 +103,7 @@ CFLAGS += $(MJ_COMMON_FLAGS)
 CXXFLAGS += $(MJ_COMMON_FLAGS)
 
 # Dependencies that all gtest based unittests should have.
-UNITTEST_LIBS := -lcap
+UNITTEST_LIBS := -l:libcap.a
 UNITTEST_DEPS := testrunner.o test_util.o
 
 USE_SYSTEM_GTEST ?= no
@@ -137,13 +137,13 @@ tests: TEST(CXX_BINARY(libminijail_unittest)) \
 	TEST(CXX_BINARY(util_unittest)) \
 	TEST(CXX_BINARY(config_parser_unittest))
 
-CC_BINARY(minijail0): LDLIBS += -lcap -ldl
+CC_BINARY(minijail0): LDLIBS += -l:libcap.a -ldl
 CC_BINARY(minijail0): $(CORE_OBJECT_FILES) \
 	elfparse.o minijail0.o minijail0_cli.o
 clean: CLEAN(minijail0)
 
 
-CC_LIBRARY(libminijail.so): LDLIBS += -lcap
+CC_LIBRARY(libminijail.so): LDLIBS += -l:libcap.a
 CC_LIBRARY(libminijail.so): $(CORE_OBJECT_FILES)
 clean: CLEAN(libminijail.so)
 
@@ -164,7 +164,7 @@ clean: CLEAN(libminijail_unittest)
 TEST(CXX_BINARY(libminijail_unittest)): CC_LIBRARY(libminijailpreload.so)
 
 
-CC_LIBRARY(libminijailpreload.so): LDLIBS += -lcap -ldl
+CC_LIBRARY(libminijailpreload.so): LDLIBS += -l:libcap.a -ldl
 CC_LIBRARY(libminijailpreload.so): libminijailpreload.o $(CORE_OBJECT_FILES)
 clean: CLEAN(libminijailpreload.so)
 
-- 
2.38.1

